modules/scratchFetch.ts

```diff
@@ -0,0 +1,242 @@
+			).catch(() => {});
+			message.reply({
+				embeds: [
+					{
+						title: studiodata.title,
+						description: long(studiodata.description, 400, "..."),
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `comments`,
+								value: studiodata.comments_allowed
+									? `${studiodata.stats.comments}`
+									: `${studiodata.stats.comments} (off)`,
+								inline: true,
+							},
+							{
+								name: `followers`,
+								value: studiodata.stats.followers,
+								inline: true,
+							},
+							{
+								name: `managers`,
+								value: studiodata.stats.managers,
+								inline: true,
+							},
+							{
+								name: `projects`,
+								value: studiodata.stats.projects,
+								inline: true,
+							},
+						],
+						thumbnail: {
+							url: studiodata.image,
+						},
+
+						footer: {
+							text: notSet ? "Disable this using /settings" : "",
+						},
+						url: `https://scratch.mit.edu/studios/${urlParts[4]}`,
+					},
+				],
+				allowedMentions: { repliedUser: false },
+			});
+
+			break;
+		case "discuss":
+			switch (urlParts[4]) {
+				case "topic":
+					const topicdata = await fetchApiData(
+						`https://scratchdb.lefty.one/v3/forum/topic/info/${urlParts[5]}/`,
+					).catch(() => {});
+					const topicposts = await fetchApiData(
+						`https://scratchdb.lefty.one/v3/forum/topic/posts/${urlParts[5]}/`,
+					).catch(() => {});
+					const topicfirst = topicposts[0];
+					message.reply({
+						embeds: [
+							{
+								title: topicdata.title,
+								description:
+									topicfirst.content
+										.bb /*`**In**: ${topicdata.category} ${topicdata.closed=0 ? "closed":""}`*/,
+
+								fields: [
+									{
+										name: "posts",
+										value: topicdata.post_count,
+									},
+								],
+								footer: {
+									text: notSet ? "Disable this using /settings" : "",
+								},
+								author: {
+									name: topicfirst.username,
+									url: `https://scratch.mit.edu/users/${topicfirst.username}`,
+								},
+								url: `https://scratch.mit.edu/discuss/topic/${urlParts[5]}`,
+							},
+						],
+						allowedMentions: { repliedUser: false },
+					});
+
+					break;
+				case "post":
+					const postdata = await fetchApiData(
+						`https://scratchdb.lefty.one/v3/forum/post/info/${urlParts[5]}/`,
+					).catch(() => {});
+					message.reply({
+						embeds: [
+							{
+								title: postdata.title,
+								description: postdata.content.bb,
+								color: 0x885cd4,
+
+								footer: {
+									text: notSet ? "Disable this using /settings" : "",
+								},
+								author: {
+									name: postdata.username,
+									url: `https://scratch.mit.edu/users/${postdata.username}`,
+								},
+								url: `https://scratch.mit.edu/discuss/post/${urlParts[5]}`,
+							},
+						],
+						allowedMentions: { repliedUser: false },
+					});
+					break;
+				default:
+					break;
+			}
+			break;
```

Comment:
```suggestion
			}
```
everything else here doesn't do anything